packages/ui-components/src/components/ToggleButton.tsx

```diff
@@ -1,198 +1,88 @@
-            icon = props.icon;
-            toggleButton.value = "";
-        }
-        if (props.value && props.valueHasLatex) {
-            toggleButton.value = (
-                <MathJax hideUntilTypeset={"first"} inline dynamic>
-                    {toggleButton.value}
-                </MathJax>
-            ) as any as string;
-        }
+    // We use `value` or `children` to display the button content.
+    if (props.value != null && props.children != null) {
+        console.warn(
+            "Button: Both `value` and `children` props are provided. Using `value`.",
+        );
     }
-
-    if (isSelected === true) {
-        if (!props.disabled) {
-            if (!props.alert) {
-                toggleButton.backgroundColor = "var(--mainBlue)";
-            } else {
-                toggleButton.backgroundColor = "var(--mainRed)";
-            }
-            toggleButton.color = "var(--canvas)";
-            if (props.switch_value) toggleButton.value = props.switch_value;
-        }
+    let value = props.value ?? props.children;
+    if (value && props.valueHasLatex) {
+        value = (
+            <MathJax hideUntilTypeset={"first"} inline dynamic>
+                {value}
+            </MathJax>
+        );
     }
 
-    function handleClick() {
-        if (props.onClick)
-            props.onClick(
-                props.index !== null && props.index !== undefined
-                    ? props.index
-                    : null,
-            );
-    }
+    let icon = props.icon && <div className="button-icon">{props.icon}</div>;
 
-    if (props.label) {
-        label.value = props.label;
-    }
-
-    if (props.width) {
-        if (props.width === "menu") {
-            toggleButton.width = "var(--menuWidth)";
-            if (props.label) {
-                container.width = "var(--menuWidth)";
-                toggleButton.width = "100%";
-            }
-        }
-    }
-
-    if (props.num === "first") {
-        toggleButton.borderRadius = "5px 0px 0px 5px";
-    }
-
-    if (props.num === "last") {
-        toggleButton.borderRadius = "0px 5px 5px 0px";
-    }
-
-    if (props.num === "first_vert") {
-        toggleButton.borderRadius = "5px 5px 0px 0px";
-    }
-
-    if (props.num === "last_vert") {
-        toggleButton.borderRadius = "0px 0px 5px 5px";
-    }
+    const Input = inputType === "radio" ? Radio : Checkbox;
 
     return (
-        <div style={container}>
-            <p id="toggle-button-label" style={label}>
-                {label.value}
-            </p>
-            <Button
-                aria-labelledby="toggle-button-label"
-                aria-pressed={props.isSelected}
-                aria-disabled={props.disabled ? true : false}
+        <label
+            className={classNames(
+                "doenet-toggle-button-label",
+                "doenet-button",
+                "action-button",
+            )}
+        >
+            <Input
+                value=""
```

Comment:
The cypress test "boolean input as toggle button" from `booleaninput.cy.js` is now failing as it cannot click the toggle button (given by its id) because the button is not visible. 

The message is:

> <input aria-checked="true" id="\/bi_input" class="doenet-toggle-button" type="checkbox" value="" checked="">
> 
> This element <input#\/bi_input.doenet-toggle-button> is not visible because it has an effective width and height of: 0 x 0 pixels.
> 
> Fix this problem, or use {force: true} to disable error checking.

Changing lines 88 and 107 in that file to `click({ force: true })` as suggested does make the test pass again. (We often have to use that `force: true` trick to make cypress test pass, so this workaround isn't unusual.)

Reply:
I changed togglebuttons to inputs instead of buttons. Maybe that is the reason?

Reply:
Ah. I just didn't read the error. Yeah, I hid the actual inputs and they are activated by clicking on the label instead.

Reply:
Another strategy to get the cypress test to pass, then, could be to move `props.id` to be the id of the label.